.github/workflows/emscripten.yml

```diff
@@ -18,51 +18,68 @@ permissions:
 
 jobs:
   build-wasm-emscripten:
+    name: Build NumPy distribution for Pyodide
     runs-on: ubuntu-22.04
-    if: false  # NOTE: job disabled, it needs to be moved to Meson (see gh-24603)
-    # if: "github.repository == 'numpy/numpy'"
+    # To enable this workflow on a fork, comment out:
+    if: github.repository == 'numpy/numpy'
     env:
-      PYODIDE_VERSION: 0.23.1
+      PYODIDE_VERSION: 0.25.0
       # PYTHON_VERSION and EMSCRIPTEN_VERSION are determined by PYODIDE_VERSION.
       # The appropriate versions can be found in the Pyodide repodata.json
       # "info" field, or in Makefile.envs:
       # https://github.com/pyodide/pyodide/blob/main/Makefile.envs#L2
-      PYTHON_VERSION: 3.11.2
-      EMSCRIPTEN_VERSION: 3.1.32
+      PYTHON_VERSION: 3.11.3
+      EMSCRIPTEN_VERSION: 3.1.46
       NODE_VERSION: 18
     steps:
-      - name: Checkout numpy
+      - name: Checkout NumPy
         uses: actions/checkout@b4ffde65f46336ab88eb53be808477a3936bae11 # v4.1.1
         with:
-          submodules: true
-          # versioneer.py requires the latest tag to be reachable. Here we
-          # fetch the complete history to get access to the tags.
-          # A shallow clone can work when the following issue is resolved:
-          # https://github.com/actions/checkout/issues/338
-          fetch-depth: 0
-
-      - name: set up python
+          submodules: recursive
+          # The fetch-tags input shall fetch tags for versioneer.py
+          # without the need to fetch the entire history, see
+          # https://github.com/actions/checkout#usage
+          fetch-tags: true
```

Comment:
Nice change! This seems to save about 15 seconds of job runtime and a whole bunch of data transfer. We have lots of other jobs using `fetch-depth: 0`. Do you want to open a separate PR to make this change in all jobs?

Reply:
Sure!

Reply:
Addressed in gh-25900